components/movementsensor/replay/replay.go

```diff
@@ -0,0 +1,450 @@
+		return r3.Vector{}, errors.New("linear acceleration is not supported, add linear_acceleration_supported to the properties to enable")
+	}
+
+	data, err := replay.getDataFromCache(ctx)
+	if err != nil {
+		return r3.Vector{}, err
+	}
+
+	replay.currentCacheStatus["linear_acceleration"] = true
+
+	return data.linearAcceleration, nil
+}
+
+// CompassHeading returns the next compass heading from the cache.
+func (replay *replayMovementSensor) CompassHeading(ctx context.Context, extra map[string]interface{}) (float64, error) {
+	replay.mu.Lock()
+	defer replay.mu.Unlock()
+	if replay.closed {
+		return 0, errors.New("session closed")
+	}
+
+	if !replay.properties.CompassHeadingSupported {
+		return 0, errors.New("compass heading is not supported, add compass_heading_supported to the properties to enable")
+	}
+
+	data, err := replay.getDataFromCache(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	replay.currentCacheStatus["compass_heading"] = true
+
+	return data.compassHeading, nil
+}
+
+// Orientation returns the next orientation from the cache.
+func (replay *replayMovementSensor) Orientation(ctx context.Context, extra map[string]interface{}) (spatialmath.Orientation, error) {
+	replay.mu.Lock()
+	defer replay.mu.Unlock()
+	if replay.closed {
+		return nil, errors.New("session closed")
+	}
+
+	if !replay.properties.OrientationSupported {
+		return nil, errors.New("orientation is not supported, add orientation_supported to the properties to enable")
+	}
+
+	data, err := replay.getDataFromCache(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	replay.currentCacheStatus["orientation"] = true
+
+	return data.orientation, nil
+}
+
+// Properties returns the available properties for the given replay movement sensor.
+func (replay *replayMovementSensor) Properties(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+	return &replay.properties, nil
+}
+
+// Accuracy is currently not defined for replay movement sensors. TODO: CHECK THIS
+func (replay *replayMovementSensor) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	return map[string]float32{}, movementsensor.ErrMethodUnimplementedAccuracy
+}
+
+// Close stops replay movement sensor, closes the channels and its connections to the cloud.
+func (replay *replayMovementSensor) Close(ctx context.Context) error {
+	replay.mu.Lock()
+	defer replay.mu.Unlock()
+
+	replay.closed = true
+	replay.closeCloudConnection(ctx)
+	return nil
+}
+
+// TODO UNIMPLEMENTED
+func (replay *replayMovementSensor) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+	return nil, nil
+}
+
+// Reconfigure finishes the bring up of the replay movement sensor by evaluating given arguments and setting up the required cloud
+// connection.
+func (replay *replayMovementSensor) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
```

Comment:
[note] Mostly same as replay camera, Properties (line 363) and currentCacheStatus (line 360) are new additions and the filter no longer has a mime_type (line 375)